website/content/platform/development/contributing/how-to/add_endpoint_to_existing_provider.md

```diff
@@ -0,0 +1,802 @@
+    result = requests.get(url, headers=headers, timeout=10)
+    assert isinstance(result, requests.Response)
+    assert result.status_code == 200
+```
+
+#### Python
+
+The `@parameterize` section can be copied directly to the Python integration test.
+
+```console
+~/OpenBBTerminal/openbb_platform/extensions/currency/integration/test_currency_python.py
+```
+
+```python
+@parametrize(
+    "params",
+    [
+        (
+            {
+                "provider": "fmp",
+                "base": "USD,XAU",
+                "counter_currencies": "EUR,JPY,GBP",
+                "quote_type": "indirect",
+            }
+        ),
+    ],
+)
+@pytest.mark.integration
+def test_currency_snapshots(params, obb):
+    result = obb.currency.snapshots(**params)
+    assert result
+    assert isinstance(result, OBBject)
+    assert len(result.results) > 0
+```
+
+Now run `pytest` for both of these files.
+
+
+Step 6 & 7 are done.
+  - [X] Add unit test.
+  - [X] Add integration tests.
+
+## Submit A Pull Request
+
+We're already on the correct branch, `feature/currency-snapshots`, but it may be out-of-sync with the `develop` branch. Let's update it just to be sure.
+
+```console
+git fetch
+git pull origin develop
+```
+
+### Linters
+
+Before opening a pull request, run the linters over all files that were touched.
+
+- black
+- ruff
+- mypy
+- pylint
+
+Fix all items, and valid fixes for `pylint` can be disabling on that line. It won't always know what is contextually correct.
+
+Here are all the files we touched in this process:
+
+- `/OpenBBTerminal/openbb_platform/core/openbb_core/provider/standard_models/currency_snapshots.py`
+- `/OpenBBTerminal/openbb_platform/providers/fmp/openbb_fmp/models/currency_snapshots.py`
+- `/OpenBBTerminal/openbb_platform/providers/fmp/tests/test_fmp_fetchers.py`
+- `/OpenBBTerminal/openbb_platform/providers/fmp/tests/record/test_fmp_currency_snapshots_fetcher.yaml`
+- `/OpenBBTerminal/openbb_platform/extensions/currency/openbb_currency/currency_router.py`
+- `/OpenBBTerminal/openbb_platform/extensions/currency/integration/test_currency_api.py`
+- `/OpenBBTerminal/openbb_platform/extensions/currency/integration/test_currency_python.py`
+- `/OpenBBTerminal/openbb_platform/openbb/assets/module_map.json`
+- `/OpenBBTerminal/openbb_platform/openbb/package/currency.py`
+
+### Install Pre-Commit Hooks
+
+The pre-commit hooks will run the testing suite locally before the commit is made. Install them from the root of the GitHub project folder, `OpenBBTerminal`.
+
+```console
+pre-commit install
+```
+
+### Commit Changes
+
+Add all the files then commit the results to the local branch.
+
+```console
+git commit -m "add obb.currency.snapshots() endpoint and create new standard model"
+```
+
+### Push Changes
+
+Assuming the commit is successful, push the changes to the remote branch.
+
+```console
+git push --set-upstream origin feature/currency-snapshots
+```
+
+### Open a Pull Request
```

Comment:
The PR parts are repeating across all examples. Would it make sense to make them its own page and just link to it as the final step?

Reply:
They are *slightly* different, there is another page for GitHub, so really it's just to be overly descriptive. Those two are the only places where that repetition exists.